tests/ordering/tests.py

```diff
@@ -203,3 +204,68 @@ def test_order_by_fk_attname(self):
             ],
             attrgetter("headline")
         )
+
+    def test_order_by_f_expression(self):
+        self.assertQuerysetEqual(
+            Article.objects.order_by(F('headline')), [
+                "Article 1",
+                "Article 2",
+                "Article 3",
+                "Article 4",
+            ],
+            attrgetter("headline")
+        )
+
+        self.assertQuerysetEqual(
+            Article.objects.order_by(F('headline').asc()), [
+                "Article 1",
+                "Article 2",
+                "Article 3",
+                "Article 4",
+            ],
+            attrgetter("headline")
+        )
+
+        self.assertQuerysetEqual(
+            Article.objects.order_by(F('headline').desc()), [
+                "Article 4",
+                "Article 3",
+                "Article 2",
+                "Article 1",
+            ],
+            attrgetter("headline")
+        )
+
+    def test_order_by_f_expression_duplicates(self):
+        """
+        A column may only be included once (the first occurance) so we check
+        to ensure there are no duplicates by inspecting the SQL.
+        """
+        qs = Article.objects.order_by(F('headline').asc(), F('headline').desc())
+        sql = qs.query.get_compiler(qs.db).as_sql()[0].upper()
```

Comment:
`str(qs.query)` just does the above internally. We can change the older test and this to use str() though.